packages/ui/VirtualList/VirtualListBasic.js

```diff
@@ -407,7 +407,7 @@ class VirtualListBasic extends Component {
 			prevProps.overhang !== this.props.overhang ||
 			prevProps.spacing !== this.props.spacing ||
 			!equals(prevProps.itemSize, this.props.itemSize) ||
-			prevProps.itemSizes?.reduce((acc, cur) => acc + cur, 0) !== this.props.itemSizes?.reduce((acc, cur) => acc + cur, 0)
+			!shallowEqual(prevProps.itemSizes, this.props.itemSizes)
```

Comment:
This change looks better than before as we are comparing shallowEqual which we can early exit if the array objects are identical. 
I just want to check one thing, if the itemSize array doesn't change, when entering the componentDidUpdate function(eg. force update), check if the itemSize array objects are the same so that it is really possible to exit early from shallowcompare.